pkg/kube/activity.go

```diff
@@ -171,23 +188,36 @@ func (k *PipelineActivityKey) GetOrCreate(activities typev1.PipelineActivityInte
 		},
 		Spec: v1.PipelineActivitySpec{},
 	}
-	if activities == nil {
+	activitiesClient := jxClient.JenkinsV1().PipelineActivities(ns)
+
+	if activitiesClient == nil {
 		log.Warn("Warning: no PipelineActivities client available!")
 		return defaultActivity, create, nil
 	}
-	a, err := activities.Get(name, metav1.GetOptions{})
+	a, err := activitiesClient.Get(name, metav1.GetOptions{})
 	if err != nil {
 		create = true
 		a = defaultActivity
 	}
 	oldSpec := a.Spec
+	oldLabels := a.Labels
+
+	// lazy create missing sourcerepository entries
+	if a.Labels[v1.LabelSourceRepository] == "" {
+		lazyUpdateSourceRepositoryCRD(jxClient, ns, a)
+	}
+
 	updateActivity(k, a)
 	if create {
-		answer, err := activities.Create(a)
+		answer, err := activitiesClient.Create(a)
 		return answer, true, err
 	} else {
 		if !reflect.DeepEqual(&a.Spec, &oldSpec) {
-			answer, err := activities.Update(a)
+			answer, err := activitiesClient.Update(a)
+			return answer, false, err
+		}
+		if !reflect.DeepEqual(&a.Labels, &oldLabels) {
```

Comment:
The contents of these if blocks are the same, so to my eyes it reads easier if you have `!reflect.DeepEqual(&a.Spec, &oldSpec) || !reflect.DeepEqual(&a.Labels, &oldLabels)` and a single conditional